### DISCUSSION TITLE
TCP traffic port

### DISCUSSION THREAD

**Discussion by kamnev999** [postalserver/postal#3254] [Help with using Postal]
Good afternoon.
Tell me where and how can I change the outgoing TCP traffic port (port 25) to port 588?

**Comment by willpower232** [comment]
I'm going to presume it is somewhere in here but not really sure to be honest

https://github.com/postalserver/postal/blob/fd3c7ccdf6dc4ee0a76c9523cbd735159e4b8000/app/lib/smtp_client/endpoint.rb#L54

**Reply by kamnev999** [reply]
I can't find this path

**Reply by willpower232** [reply]
It will be inside the docker container

**Reply by kamnev999** [reply]
def start_smtp_session(source_ip_address: nil, allow_ssl: true)
     # @smtp_client = Net::SMTP.new(@ip_address, @server.port)
     @smtp_client = Net::SMTP.new(@ip_address, 588)

      @smtp_client.open_timeout = Postal::Config.smtp_client.open_timeout
      @smtp_client.read_timeout = Postal::Config.smtp_client.read_timeout
      @smtp_client.tls_hostname = @server.hostname
      
      
 after that, I restarted POSTAL.

It's still trying to send via port 25
Are there any other solutions?

**Reply by willpower232** [reply]
If the code change is definitely still there after you restarted Postal then I'm not sure what the solution is.

I don't know if you can forward a port internally using iptables for outgoing traffic. I know you can do it for incoming.

**Reply by kamnev999** [reply]
When redirecting via iptables, it still tries to send via port 25
